File: src/data/types/craftAction.ts
import type { ResourceInstance } from "./resourceInstance.ts";
import type { Recipe } from "./recipe.ts";
import type { TierColor } from "./tierColor.ts";

// Necessary completion: Agent 16's contract passes a "craftAction" to
// assignToCraft()/resolveBackgroundCrafting() but never defines its shape.
// Bundles everything craft() needs besides the crafter's own tier (which
// comes from the CrewMember doing the work, not from this action): the
// consumed inputs, which recipe, and which schematic. `id` is what
// CrewMember.assignedCraftId points at once assigned.
export interface CraftAction {
  id: string;
  inputs: ResourceInstance[];
  recipe: Recipe;
  schematicTier: TierColor;
}

File: unity/ProfitableCore/Schema/CraftAction.cs
namespace Profitable.Core.Schema;

// Ports src/data/types/craftAction.ts. Bundles everything Crafter.Craft
// needs besides the crafter's own tier (which comes from the CrewMember
// doing the work, not from this action): the consumed inputs, which
// recipe, and which schematic. Id is what CrewMember.AssignedCraftId
// points at once assigned.
public class CraftAction
{
    public string Id { get; set; } = string.Empty;
    public List<ResourceInstance> Inputs { get; set; } = new();
    public Recipe Recipe { get; set; } = new();
    public TierColor SchematicTier { get; set; }
}
